samcli/lib/cli_validation/remote_invoke_options_validations.py

```diff
@@ -48,10 +46,9 @@ def wrapped(*args, **kwargs):
 
         validator.validate()
 
-        # if no event nor event_file arguments are given, read from stdin
-        if not event and not event_file:
-            LOG.debug("Neither --event nor --event-file options have been provided, reading from stdin")
-            kwargs["event_file"] = cast(TextIOWrapper, sys.stdin)
+        # If "-" is provided for --event-file, click uses it as a special file to refer to stdin.
+        if event_file and event_file.fileno() == sys.stdin.fileno():
```

Comment:
just checking: is this the same behavior as local?

Reply:
Yes, `sam local` behaves in a similar way when `-` is passed to `-e` file path option. Click intelligently treats this as being read from `stdin` as the file stream.